client/src/components/PrivateRoute.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { useContext } from 'react'
+import { Route } from 'react-router-dom'
+import { useKeycloak } from '@react-keycloak/web'
+import { AppContext } from '../AppContext';
+import { Loading } from './Loading';
+import { useNetworkStatus } from 'react-offix-hooks';
+
+export const PrivateRoute: React.FC<any> = ({ component: Component, ...rest }) => {
+  const isOnline = useNetworkStatus();
+  const [keycloak, initialized] = useKeycloak();
+  const { keycloakEnabled } = useContext(AppContext);
```

Comment:
[Opiniated] [Optional] I know this might not look cool, but we could have keycloakEnabled coming from helper import rather than context. We know if keycloak is enabled based on static config. Keycloak lib was written long time ago and it is working in very basic JQuery Like way. Basically you could include keycloak into your script
and it will look for keycloak.json in your root path. All stuff on global scope.